internal/venice-client-common/src/main/java/com/linkedin/venice/controllerapi/D2ServiceDiscoveryResponse.java

```diff
@@ -14,9 +15,17 @@ public void setD2Service(String d2Service) {
     this.d2Service = d2Service;
   }
 
+  public String getServerD2Service() {
```

Comment:
I see that this field is only being set when cluster discovery is called on routers. Can you also make controllers set this field?

Reply:
Made the changes such that it's being set in `ClusterDiscovery`. Added a new method `getServerD2Service(String clusterName)` to `VeniceHelixAdmin` to help with this